lib/db/schema/patch-092-093.sql

```diff
@@ -0,0 +1,47 @@
+-- This migration removes the use of `ROW_COUNT()` on the last
+-- remaining procedures.
+
+SET NAMES utf8mb4 COLLATE utf8mb4_bin;
+
+CALL assertPatchLevel('92');
+
+-- Update to not use `ROW_COUNT()`
+CREATE PROCEDURE `setPrimaryEmail_4` (
+  IN `inUid` BINARY(16),
+  IN `inNormalizedEmail` VARCHAR(255) CHARACTER SET utf8 COLLATE utf8_bin
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+     UPDATE emails SET isPrimary = false WHERE uid = inUid AND isPrimary = true;
+     UPDATE emails SET isPrimary = true WHERE uid = inUid AND isPrimary = false AND normalizedEmail = inNormalizedEmail;
```

Comment:
The previous version of this procedure used `ROW_COUNT` to check whether the user actually owns `inNormalizedEmail`, to avoid accidentally leaving them in a state where they do now have a primary emil address:

```
IF @updateCount = 0 THEN
  SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 1062, MESSAGE_TEXT = 'Can not change email. Could not find email.';
END IF;
```

Does the responsibility for this check now live somewhere else?

Reply:
@rfk The responsibility of this check lives on the auth-server, https://github.com/mozilla/fxa-auth-server/blob/42d165eccc1ab0694e9cc46ab40362a7aa8e393d/lib/routes/emails.js#L759. I'll add some additional comments to better document these.

Reply:
Aha, thanks!

There's a pedantic part of me that doesn't quite feel comfortable that this check lives outside of the transaction boundary.  You can imagine a (highly unlikely) sequence of events where concurrent calls to `deleteEmail` and `setPrimaryEmail` result in an account with no primary email address.  Such calls shouldn't happen concurrently in practice, but "shouldn't" is not the same as "can't"...

Reply:
Changing primary email happens infrequently enough that I feel like we could spare an additional db lookup (say, to confirm that the user owns the given email address at the beginning of the procedure) in order to guard against producing a bad db state.